scripts/asm-installer/install_asm

```diff
@@ -1735,28 +1735,38 @@ get_cluster_labels() {
 }
 
 is_cluster_registered() {
-
   if ! is_membership_crd_installed; then
     false
     return
   fi
 
-  local WANT
-  WANT="//container.googleapis.com/projects/${PROJECT_ID}/locations/${CLUSTER_LOCATION}/clusters/${CLUSTER_NAME}"
-  local LIST; LIST="$(gcloud container hub memberships list --project "${PROJECT_ID}" --format=json | grep "${WANT}")"
-
-  if [[ -z "${LIST}" ]]; then
-    false
-    return
-  fi
-
   local IDENTITY_PROVIDER
   IDENTITY_PROVIDER="$(retry 2 kubectl get memberships.hub.gke.io \
     membership -ojson 2>/dev/null | jq .spec.identity_provider)"
 
-  if [[ -z "${IDENTITY_PROVIDER}" ]] || [[ "${IDENTITY_PROVIDER}" = 'null' ]]; then
+  if [[ -z "${IDENTITY_PROVIDER}" ]] || [[ "${IDENTITY_PROVIDER}" == 'null' ]]; then
     false
   fi
+
+  populate_environ_info
+
+  if [[ "${PROJECT_ID}" == "${ENVIRON_PROJECT_ID}" ]]; then
+    local WANT
+    WANT="//container.googleapis.com/projects/${PROJECT_ID}/locations/${CLUSTER_LOCATION}/clusters/${CLUSTER_NAME}"
```

Comment:
Can you just replace PROJECT_ID with ENVIRON_PROJECT_ID and remove the `--project "${PROJECT_ID}"` flag?

Reply:
How about permission check? Try to check membership, if permission issue, then warn?

Reply:
Makes sense to me. We can just make it a general "Failed to validate environ registration" warning.

Reply:
done.